### ISSUE TITLE
[BUG] read CSV corrupting data when floating point number in exponent format

### ISSUE THREAD

**Issue by MurrayData** [rapidsai/cudf#987]
I read a CSV file of 6 million records including latitude/Longitude points captured by food hygiene inspectors using handheld mobile devices. As UK & Ireland, this data was in the range 48 to 60 latitude, -8 to 3 longitude.

My Spatia software threw an error while performing a coordinate conversion process on the points due to out of range values. As the data had been pre-filtered to remove outliers, this puzzled me. 

I investigated further and found that read csv is not working correctly when a floating point number is written in exponent format e.g. -9e-06 is being converted to -142706.0.

The csv file was exported from a MySQL database using their standard utilities which uses exponent format for small absolute values.

I extracted all such records to a separate CSV file (145 records):

```
ID,latitude,longitude
26994,50.87296818,-3.324e-05
34499,51.68672991,-6.706e-05
34849,51.687048,-9e-06
35083,51.743898,9e-06
93871,53.363091,-9e-06
94482,53.362668,-2.9e-05
94857,53.363091,-9e-06
97159,52.746118,8.1e-05
153395,51.13764,8.6e-05
153997,51.13764,8.6e-05
```
When read into a CUDF dataframe:
```
%%time
fdf = cudf.read_csv(fsa_file,names=names,dtype=dtypes,skiprows=1)

CPU times: user 4.3 ms, sys: 16.1 ms, total: 20.4 ms
Wall time: 18.5 ms
```
results are:
```
print(fdf)

       ID                lat                 lon
 0  26994        50.87296818          -3.3292705
 1  34499 51.686729910000004          -6.7112705
 2  34849          51.687048           -142706.0
 3  35083          51.743898            142706.0
 4  93871          53.363091           -142706.0
 5  94482          53.362668 -3.4270499999999995
 6  94857          53.363091           -142706.0
 7  97159  52.74611800000001   8.627049999999999
 8 153395           51.13764   9.127049999999999
 9 153997           51.13764   9.127049999999999
[135 more rows]
```

The actual CSV file extract use is attached (zipped)
[fhrs_lat_lon_20190217_exponent.zip](https://github.com/rapidsai/cudf/files/2879577/fhrs_lat_lon_20190217_exponent.zip)

I am using the 0.5-cuda10.0-runtime-ubuntu16.04-gcc5-py3.6 version of the Docker container.

**Comment by j-ieong** [comment]
Hi @MurrayData, could you try the branch-0.6 code if possible?
Scientific notation handling was added in PR #780 and will be in the v0.6 release.

**Comment by MurrayData** [comment]
Yes, I'll try it in branch-0.6 and report back.

**Comment by mjsamoht** [comment]
Using the attached csv file from @MurrayData I was able to reproduce two problems with our handling of scientific data:
1. We incorrectly handle **negative exponents that start with a zero**.  E.g. 3.1e-01 will be interpreted as 3.1e-1 because we multiply the sign with the first digit of the exponent.  To fix this we probably want to remember the sign of the exponent and multiply at the end.
2.  Automatic data type inference will incorrectly set the type to date format.  It seems that in `dataTypeDetection()` we don't consider the case where there is an 'e' in the field so it will not be identified as float.

**Comment by mjsamoht** [comment]
@j-ieong Can we fix this for release v0.6?
